temporalio/lib/temporalio/contrib/open_telemetry.rb

```diff
@@ -86,10 +86,24 @@ def _with_started_span(
           attributes: nil,
           outbound_input: nil
         )
-          tracer.in_span(name, attributes:, kind:) do
+          # We cannot use tracer.in_span because it always assumes we want to set the status as error on exception, but
+          # we do not want to do that for benign exceptions. This is effectively a copy of the source of in_span with
+          # that change.
+          span = nil
+          span = tracer.start_span(name, attributes:, kind:)
+          ::OpenTelemetry::Trace.with_span(span) do
             _apply_context_to_headers(outbound_input.headers) if outbound_input
             yield
           end
+        rescue Exception => e # rubocop:disable Lint/RescueException
+          span&.record_exception(e)
+          # Only set the status if it is not a benign application error
+          unless e.is_a?(Error::ApplicationError) && e.category == Error::ApplicationError::Category::BENIGN
+            span&.status = ::OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{e.class}")
+          end
```

Comment:
I know it's only these two spots but maybe we want to make this a helper like `record_exception_span(e)` or something that can encapsulate the rules so if we add more later or more spots later it's easier.